### PR TITLE
Watkins lesson 3 suggestions

### PR DESCRIPTION
**Pull request by jacobwatkins1**
## Changes
These are content changes and suggestions to lesson 3 of the Understanding Quantum Information and Computation course. Text has been added in various places to give more context and motivation whenever I thought it to be beneficial. Smaller changes such as wording have also been made. All suggestions I have made are necessarily subjective, and not all should necessarily be implemented.

## Implementation details
Changes made were primarily to the markdown writing, and some more commenting on the code sections.

## How to read this PR
Please look through the textual changes and give comments if desired, or simply accept/reject as appropriate.

### CODE REVIEW

File: notebooks/basics/quantum-circuits.ipynb [hunk 1]
```diff
@@ -22,30 +22,43 @@
...
+    }
+   },
    "cell_type": "markdown",
    "id": "adfa4cb5",
    "metadata": {},
    "source": [
     "### 1.1 Boolean circuits\n",
     "\n",
-    "Here is an example of a (classical) Boolean circuit, where the wires carry binary values and the gates represent Boolean logic operations:\n",
+    "To illustrate these ideas, let's begin with a simple example. A *Boolean circuit* is one in which the output is a single classical bit (though the input could be $n$ bits). Here is an example:\n",
```
Comment (frankharkins) [inline notebooks/basics/quantum-circuits.ipynb:48]:
```suggestion
    "To illustrate these ideas, let's begin with a simple example. A *Boolean circuit* is one in which the output is a single classical bit (though the input could be many bits). Here is an example:\n",
```

File: notebooks/basics/quantum-circuits.ipynb [hunk 2]
```diff
@@ -258,15 +327,29 @@
...
+    "      \\hline\n",
+    "      00 & 00\\\\\n",
+    "      01 & 11\\\\\n",
+    "      10 & 10\\\\\n",
+    "      11 & 01\n",
+    "    \\end{array}.\n",
+    "   $$\n",
     "\n",
-    "   Given that we order the qubits as $(\\mathsf{X}, \\mathsf{Y})$, the matrix representation of the controlled-NOT gate is this:\n",
+    "   Given that our circuit ordering convention, the matrix representation of the controlled-NOT gate is this:\n",
```
Comment (frankharkins) [inline notebooks/basics/quantum-circuits.ipynb:352]:
```suggestion
    "   Given that our qubit ordering convention, the matrix representation of the controlled-NOT gate is this:\n",
```

File: notebooks/basics/quantum-circuits.ipynb [hunk 3]
```diff
@@ -1019,17 +1179,19 @@
    "source": [
     "#### Implementing projective measurements using standard basis measurements\n",
     "\n",
-    "Arbitrary projective measurements can be implemented using unitary operations, standard basis measurements, and an extra workspace system, as will now be explained.\n",
+    "There is one sense in which general projective measurements are nothing new: they can be implemented using unitary operations, standard basis measurements, and an extra workspace system.\n",
```
Comment (frankharkins) [inline notebooks/basics/quantum-circuits.ipynb:1182]:
```suggestion
    "There is one sense in which general projective measurements are nothing new; they can be implemented using unitary operations, standard basis measurements, and an extra workspace system.\n",
```

**Review by frankharkins — COMMENTED** [review]
Thanks! I've left some small comments, but didn't make any suggestions to the larger changes (I'll wait until John reviews first).

File: notebooks/basics/quantum-circuits.ipynb [hunk 4]
```diff
@@ -22,30 +22,43 @@
...
+    "\n",
+    "The flow of information along the wires goes from left to right. The wires on the leftmost side of the figure, labeled $\\mathsf{X}$ and $\\mathsf{Y}$, are the input bits, which can each be set to whatever binary value we choose. The wire on the right-hand side is the output, which is completely determined by the input and cannot be specified separately. Traveling along any wire, the value (0 or 1) stays the same, only changing when passing through a gate. Wherever two wires intersect, we use a filled-in dot to denote the wires being *connected*. If two wires cross with no dot, then the wires simply pass over one another, not affecting each other in any way. Other conventions exist, and sometimes a gap or a small loop is used to indicate unconnected crossing wires.\n",
     "\n",
-    "The gates are AND gates (labeled $\\wedge$), OR gates (labeled $\\vee$), and NOT gates (labeled $\\neg$).\n",
-    "The functions computed by these gates will likely be familiar to many readers, but here they are represented by tables of values:\n",
+    "![UnconnectedWire.png](attachment:UnconnectedWire.png)\n",
```
Comment (frankharkins) [inline notebooks/basics/quantum-circuits.ipynb:56]: Did you add this image? If so, can you put it in the `notebooks/basics/images/quantum-circuits` folder, and add it to the notebook using
```markdown
![description of the image](images/quantum-circuits/UnconnectedWire.png)
```

File: notebooks/basics/quantum-circuits.ipynb (hunk 1 above)
Comment (jacobwatkins1) [reply notebooks/basics/quantum-circuits.ipynb]: Thanks Frank, I agree with all of these and made the change!

File: notebooks/basics/quantum-circuits.ipynb (hunk 2 above)
Comment (jacobwatkins1) [reply notebooks/basics/quantum-circuits.ipynb]: Changed!

File: notebooks/basics/quantum-circuits.ipynb (hunk 3 above)
Comment (jacobwatkins1) [reply notebooks/basics/quantum-circuits.ipynb]: Changed, thanks!

File: notebooks/basics/quantum-circuits.ipynb (hunk 4 above)
Comment (jacobwatkins1) [reply notebooks/basics/quantum-circuits.ipynb]: Done! Thanks very much.

**Pull request closed without merging**